packages/common/src/types.ts

```diff
@@ -120,3 +121,109 @@ export interface GethConfigOpts extends BaseOpts {
+ */
+
+export interface StorageDump {
+  [key: string]: string
+}
+
+export type AccountFields = Partial<Pick<Account, 'nonce' | 'balance' | 'storageRoot' | 'codeHash'>>
+
+export type CacheClearingOpts = {
+  /**
+   * Full cache clearing
+   * (overrides the useThreshold option)
+   *
+   * default: true
+   */
+  clear: boolean
+  /**
+   * Clean up the cache by deleting cache elements
+   * where stored comparand is below the given
+   * threshold.
+   */
+  useThreshold?: bigint
+  /**
+   * Comparand stored along a cache element with a
+   * read or write access.
+   *
+   * This can be a block number, timestamp,
+   * consecutive number or any other bigint
+   * which makes sense as a comparison value.
+   */
+  comparand?: bigint
+}
+
+export type StorageProof = {
+  key: PrefixedHexString
+  proof: PrefixedHexString[]
+  value: PrefixedHexString
+}
+
+export type Proof = {
+  address: PrefixedHexString
+  balance: PrefixedHexString
+  codeHash: PrefixedHexString
+  nonce: PrefixedHexString
+  storageHash: PrefixedHexString
+  accountProof: PrefixedHexString[]
+  storageProof: StorageProof[]
+}
+
+type Stats = {
+  cache: {
+    size: number
+    reads: number
+    hits: number
+    writes: number
+    dels: number
+  }
+  trie: {
+    reads: number
+    writes: number
+    dels: number
+  }
+}
+
+export interface CacheInterface {
+  getOrLoad(address: Address): Promise<Account | undefined>
+  flush(): Promise<void>
+  clear(cacheClearingOpts: CacheClearingOpts): void
+  put(address: Address, account: Account | undefined): void
+  del(address: Address): void
+  checkpoint(): void
+  revert(): void
+  commit(): void
+  stats(reset?: boolean): Stats
+}
+
+export interface StateAccess {
+  accountExists(address: Address): Promise<boolean>
+  getAccount(address: Address): Promise<Account | undefined>
+  putAccount(address: Address, account: Account): Promise<void>
+  deleteAccount(address: Address): Promise<void>
+  modifyAccountFields(address: Address, accountFields: AccountFields): Promise<void>
+  putContractCode(address: Address, value: Buffer): Promise<void>
+  getContractCode(address: Address): Promise<Buffer>
+  getContractStorage(address: Address, key: Buffer): Promise<Buffer>
+  putContractStorage(address: Address, key: Buffer, value: Buffer): Promise<void>
+  clearContractStorage(address: Address): Promise<void>
+  checkpoint(): Promise<void>
+  commit(): Promise<void>
+  revert(): Promise<void>
+  getStateRoot(): Promise<Buffer>
+  setStateRoot(stateRoot: Buffer, cacheClearingOptions?: CacheClearingOpts): Promise<void>
+  getProof?(address: Address, storageSlots: Buffer[]): Promise<Proof>
+  verifyProof?(proof: Proof): Promise<boolean>
+  hasStateRoot(root: Buffer): Promise<boolean>
+}
+
+export interface StateManagerInterface extends StateAccess {
+  cache?: CacheInterface
+  copy(): StateManagerInterface
+  flush(): Promise<void>
+  dumpStorage(address: Address): Promise<StorageDump>
+}
```

Comment:
I think we should be really sparse with taking over interfaces to Common and not "throwing things all over here".

So these should really only be the main interfaces, like `StateManagerInterface`, eventually `BlockchainInterface`, and not much else.

So these should be very clear, very well-managed interfaces, where there is a lot of attention onto. Otherwise I fear that we will likely overload and produce too much complexity which won't make things easier to oversee respectively more stable.

I would particularly suggest that we do not expose the cache so prominently, I guess I added this (right?), but anyhow, it would likely be better if we just hack this a bit on the very few direct cache accesses we have internally than build up such heavy structures like this.

I would suggest the following:

- Remove cache from SM Interface, delete all related interfaces
- (CacheClearingOpts is outdated anyhow)
- Remove the `dumpStorage` and `StorageDump` from the interface in general, this is only used in one very obscure client location, we can hack this in there
- Throw the `StateAccess` and `StateManagerInterface` interfaces together (never gotten a big fan of this)
- Definitely remove this `AccountFields` partial
- Eventually keep the Proof types, but really only if stable enough

Two other things:

1. This could also very well be an own PR, totally independent from other stuff
2. I think we should rather do a dedicated file `interfaces.ts` for this